### PR TITLE
Extend SET_ATTITUDE_TARGET message for 3D thrust setpoints

### PR DESCRIPTION
**Pull request by Jaeyoung-Lim**
**Problem Description**
With recent development of fully actuated systems and more complex actuation being available on VTOL vehicles, we need a way to conveniently set 3D thrust setpoints on vehicles. 

This adds a 3D thrust setpoint in the SET_ATTITUDE_TARGET to support fully actuated vehicles

The reference implementation is provided in https://github.com/PX4/PX4-Autopilot/pull/16994

### CODE REVIEW

**Review by amilcarlucas — CHANGES_REQUESTED** [review]
What are the units?

**Comment by Jaeyoung-Lim** [comment]
@amilcarlucas It is normalized thrust therefore it is unitless. Same as the thrust

**Comment by amilcarlucas** [comment]
Then add it to the description of this message as well.

**Comment by Jaeyoung-Lim** [comment]
@amilcarlucas Done

**Comment by Jaeyoung-Lim** [comment]
@amilcarlucas Is there anything else that needs to be improved?

**Comment by amilcarlucas** [comment]
Now you need for it to get reviewed by the Team.
I do not like that it changes a stable message. I think that only wip messages should be changed.

**Comment by Jaeyoung-Lim** [comment]
@amilcarlucas This has been something that is limiting the development for a while, and since it is a stable message I also understand that messages shouldn't be changed. Therefore I added it as an extension.

If there is nothing to be changed from your side, would you be able to re-review this PR?

**Comment by amilcarlucas** [comment]
Can you provide an ardupilot reference implementation as well?

**Comment by Jaeyoung-Lim** [comment]
@amilcarlucas Unfortunately not. I am not familiar with Ardupilot. As far as I understand it is not required to provide a reference implementation for multiple autopilots.

May I ask why this is needed for re-reviewing this PR?

**Comment by amilcarlucas** [comment]
No second reference implementation is required but it woul not hurt.

**Review by amilcarlucas — APPROVED** [review]

**Comment by Jaeyoung-Lim** [comment]
@amilcarlucas Thanks!

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -3540,6 +3540,9 @@
       <entry value="4" name="ATTITUDE_TARGET_TYPEMASK_BODY_YAW_RATE_IGNORE">
         <description>Ignore body yaw rate</description>
       </entry>
+      <entry value="32" name="ATTITUDE_TARGET_TYPEMASK_THRUST_BODY_SET">
+        <description>Use 3D body thrust setpoint instead of throttle</description>
```
[the remark this answers is not in the document]
Comment (Jaeyoung-Lim) [reply message_definitions/v1.0/common.xml:3544]: @hamishwillee This is following the foot step of the `POSITION_TARGET_TYPEMASK`: https://github.com/mavlink/mavlink/blob/35b3f5e51eeb807486a1b079565e74d4e6ad3418/message_definitions/v1.0/common.xml#L3523. The problem with defining it as a ignore is that if none of the ignore flags it means that the user wants to use both as a default, which is not a valid case

[the remark this answers is not in the document]
Comment (hamishwillee) [reply message_definitions/v1.0/common.xml:3544]: OK!

**Comment by jbwillis** [comment]
I am interested in using this with ROS1 for testing a VTOL control algorithm I've been working on. @Jaeyoung-Lim has there been effort in updating MAVROS to support it?

**Comment by Jaeyoung-Lim** [comment]
@jbwillis mavros is only mapping ros topics to mavlink, which requires this PR to be merged. Therefore the answer is no.
In case for your use case, the real part missing is the mavlink integration into the firmware for VTOLS

**Comment by jbwillis** [comment]
@Jaeyoung-Lim Right, just wondering if there was some parallel development going on with MAVROS as well. I'm planning on working off of your branches on the PX4 autopilot.

**Comment by Jaeyoung-Lim** [comment]
@jbwillis Great! I will push this up to mavros once this is merged

**Comment by hamishwillee** [comment]
This is a non-breaking/compatible extension which appears to make sense. I will contact @auturgy again for comment.

**Comment by auturgy** [comment]
Looks fine. Happy to merge

**Review by LorenzMeier — APPROVED** [review]

**Pull request merged**